README.md

```diff
@@ -1,62 +1,549 @@
-hypertable
-==============================================================================
+# Hypertable
 
 To follow the early life of this project, see
 [oss-components#pn-flexbox-tables](https://github.com/upfluence/oss-components/pull/54/files)
 
+## Table of Contents
 
-Compatibility
-------------------------------------------------------------------------------
+- [Compatibility](#compatibility)
+- [Installation](#installation)
+- [Architecture](#architecture)
+- [Quick Start](#quick-start)
+- [Core Concepts](#core-concepts)
+- [Rendering System](#rendering-system)
+- [Built-in Components](#built-in-components)
+- [Events](#events)
+- [Contributing](#contributing)
 
-* Ember.js v3.24 or above
-* Ember CLI v3.24 or above
-* Node.js v12 or above
+## Compatibility
 
+- Ember.js v3.24 or above
+- Ember CLI v3.24 or above
+- Node.js v12 or above
 
-Installation
-------------------------------------------------------------------------------
+## Installation
 
 ```
 ember install hypertable
 ```
 
+## Architecture
 
-Usage
-------------------------------------------------------------------------------
+- **TableHandler** - The core system that manages data state, columns, filters, and selections
+- **TableManager** - Interface for fetching and persisting column definitions and configurations
+- **RowsFetcher** - Interface for fetching paginated data
+- **RenderingResolver** - Maps column definitions to appropriate cell, header, and filter renderers
+- **Renderers** - Modular components for displaying cells, headers, and filters
 
-[Longer description of how to use the addon in apps.]
+## Quick Start
 
+### 1. Implement Required Interfaces
 
-Contributing
-------------------------------------------------------------------------------
+Create implementations for the two required interfaces and add a rendering resolver if needed:
+
+```ts
+export default class MyController extends Controller {
+  constructor() {
+    super(...arguments);
+
+    const manager = new MyTableManager();
+    const fetcher = new MyRowsFetcher();
+    const renderingResolver = new MyRenderingResolver(); // Optional
```

Comment:
`MyTableManager`,  `MyRowsFetcher` and `MyRenderingResolver` don't exist and is a bit misleading.
Best use the default names and also add the required imports ?